### PR TITLE
Automatically add a Panel Group when adding a Panel to an empty dashboard

### PR DESCRIPTION
**Pull request by LukeTillman**
This fixes a bug where if you clicked "Add Panel" when a dashboard was empty with no panel groups, you got an error message saying "No panel groups to add a panel to". After discussing a bit with Mohit, we decided the new behavior here is to automatically add an empty panel group for the user when this happens. Here's what that looks like in action:

https://user-images.githubusercontent.com/428023/202031152-1bd1e1bc-31cf-43b4-bdd5-e09d34475d29.mov

### CODE REVIEW

**Review by juliepagano — APPROVED** [review]
One small optional suggestion, but otherwise looks good to me. 🎉

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,8 +4,9 @@
 
 - [ENHANCEMENT] User always sees edit icons in edit mode (instead of only seeing on hover) #748
 - [ENHANCEMENT] Add unit selector to time series chart options #744
-- [BUGFIX] Fix the default config file used in the docker images and in the archive #745
```
[the remark this answers is not in the document]
Comment (cndonovan) [reply CHANGELOG.md:7]: I _think_ we're ordering them as: Feature, Enhancement, Bug fix, Breaking change

File: ui/dashboards/src/context/DashboardProvider/panel-editor-slice.ts [hunk 2]
```diff
@@ -223,8 +227,13 @@ export function createPanelEditorSlice(): StateCreator<
...
-      // Open the editor with the new state
       set((state) => {
+        // Add the new panel group if one was created for the panel
+        if (newGroup !== undefined) {
+          addPanelGroup(state, newGroup);
+        }
+
+        // Open the editor with the new state
         state.panelEditor = editorState;
       });
```
Comment (cndonovan) [inline ui/dashboards/src/context/DashboardProvider/panel-editor-slice.ts:238]: Could be cool to always say either `set((state) => ...)` or `set((draft) => ...)` (I noticed we have these different argument names when calling `set` in in this file and panel-group-editor-slice)

I lean toward `state` but also it's not a big deal

File: ui/dashboards/src/context/DashboardProvider/panel-editor-slice.ts [hunk 3]
```diff
@@ -173,13 +175,15 @@ export function createPanelEditorSlice(): StateCreator<
...
+      // If a panel group isn't supplied, add to the first group or create a group if there aren't any
+      let newGroup: PanelGroupDefinition | undefined = undefined;
       if (panelGroupId === undefined) {
-        const firstGroupId = get().panelGroupOrder[0];
-        if (firstGroupId === undefined) {
-          throw new Error('No panel groups to add a panel to');
+        panelGroupId = get().panelGroupOrder[0];
+        if (panelGroupId === undefined) {
+          newGroup = createEmptyPanelGroup();
+          newGroup.title = 'Panel Group';
```
Comment (cndonovan) [inline ui/dashboards/src/context/DashboardProvider/panel-editor-slice.ts:184]: I like the idea of 'Group 1' but it's not my say in the end, just throwing it out there

**Review by cndonovan — APPROVED** [review]
yay!

**Review by eunicorn — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (LukeTillman) [reply CHANGELOG.md:7]: Yeah, basically there's an order they're supposed to be sorted in, so I was just preemptively sorting 😄

File: ui/dashboards/src/context/DashboardProvider/panel-editor-slice.ts [hunk 4]
```diff
@@ -173,13 +175,15 @@ export function createPanelEditorSlice(): StateCreator<
...
 
     openAddPanel(panelGroupId) {
-      // If a panel group isn't supplied, add to the first group
+      // If a panel group isn't supplied, add to the first group or create a group if there aren't any
+      let newGroup: PanelGroupDefinition | undefined = undefined;
       if (panelGroupId === undefined) {
-        const firstGroupId = get().panelGroupOrder[0];
-        if (firstGroupId === undefined) {
-          throw new Error('No panel groups to add a panel to');
+        panelGroupId = get().panelGroupOrder[0];
```
[the remark this answers is not in the document]
Comment (LukeTillman) [reply ui/dashboards/src/context/DashboardProvider/panel-editor-slice.ts:181]: Good idea, fixed!

[the remark this answers is not in the document]
Comment (LukeTillman) [reply ui/dashboards/src/context/DashboardProvider/panel-editor-slice.ts]: I went with the null-coalescing assignment operator which always feels fun when I get to use it. 😂

**Pull request merged**